File: Modules/Groups/Http/Resources/GroupResource.php
<?php

namespace Modules\Groups\Http\Resources;

use Illuminate\Http\Resources\Json\JsonResource;
use Modules\Auth\Http\Resources\UserResource;
use Modules\Lookups\Http\Resources\SubstanceCategoryResource;
use Modules\Sessions\Http\Resources\SessionResource;

class GroupResource extends JsonResource
{
    public function toArray($request): array
    {
        return [
            'id' => $this->id,
            'group_type' => $this->group_type,
            'status' => $this->status,
            'name_ar' => $this->name_ar,
            'name_en' => $this->name_en,
            'min_members' => $this->min_members,
            'max_members' => $this->max_members,
            'created_at' => $this->created_at,
            'updated_at' => $this->updated_at,
            'instructor' => new UserResource($this->whenLoaded('instructor')),
            'category' => new SubstanceCategoryResource($this->whenLoaded('category')),
            'members' => GroupMemberResource::collection($this->whenLoaded('members')),
            'sessions' => SessionResource::collection($this->whenLoaded('sessions')),
        ];
    }
}

File: Solid.Api/Features/Groups/GroupResource.cs
using Solid.Api.Database.Entities;

namespace Solid.Api.Features.Groups;

public static class GroupResource
{
    public static object From(Group group)
    {
        return new
        {
            id = group.Id,
            group_type = group.GroupType,
            status = group.Status,
            substance_category_id = group.SubstanceCategoryId,
            substance_category = group.SubstanceCategory is null
                ? null
                : new
                {
                    id = group.SubstanceCategory.Id,
                    name_ar = group.SubstanceCategory.NameAr,
                    name_en = group.SubstanceCategory.NameEn
                },
            name_ar = group.NameAr,
            name_en = group.NameEn,
            min_members = group.MinMembers,
            max_members = group.MaxMembers,
            members_count = group.Members.Count(member => member.IsActive),
            created_at = group.CreatedAt,
            updated_at = group.UpdatedAt
        };
    }
}
